### PR TITLE
Update stock, sync item, shopify variant id and pre-requisites

### PR DESCRIPTION
**Pull request by saurabh6790**
- Update Stock while syncing data with shopify
- create item while syncing qty if item is missing on shopify
- patch to set variant id
- pre-requisite fixes

### CODE REVIEW

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 1]
```diff
@@ -121,11 +123,13 @@ def set_new_attribute_values(item_attr, values):
 			})
 
 def create_item(item, warehouse, has_variant=0, attributes=[],variant_of=None):
+	print item.get("image", {})
```
Comment (nabinhait) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py]: remove print

**Comment by anandpdoshi** [comment]
@nabinhait are you going to look into this?

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 2]
```diff
@@ -120,25 +121,39 @@ def set_new_attribute_values(item_attr, values):
...
-		"default_warehouse": warehouse
-	}).insert()
+		"default_warehouse": warehouse,
+		"image": get_item_image(item)
+	}
+	
+	name, item_details = get_item_details(item)  
+		
+	if not name:
+		frappe.errprint(["var", item_details, item_dict])
```
Comment (anandpdoshi) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py]: This should only be used for debugging

File: erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py [hunk 3]
```diff
@@ -180,50 +195,121 @@ def get_item_group(product_type=None):
...
+		where sync_with_shopify=1 and variant_of is null""", as_dict=1):
+		sync_item_with_shopify(item, price_list, warehouse)
+		
+def sync_item_with_shopify(item, price_list, warehouse):
+	variant_item_code_list = []
 
-		if item.get("has_variants"):
-			variant_list, options, variant_item_code = get_variant_attributes(item, price_list, warehouse)
+	item_data = {
+				"product": {
```
Comment (anandpdoshi) [inline erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py]: looks like an indentation problem

File: erpnext_shopify/patches/V1_0/set_variant_id.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import frappe
+from erpnext_shopify.utils import get_request
+
+def execute():
+	shopify_settings = frappe.get_doc("Shopify Settings")
+	if shopify_settings.shopify_url:
+		for item in frappe.db.sql("""select name, item_code, shopify_id, has_variants, variant_of from tabItem 
+			where sync_with_shopify=1 and shopify_id is not null""", as_dict=1):
+			
+			if item.get("varint_of"):
```
Comment (anandpdoshi) [inline erpnext_shopify/patches/V1_0/set_variant_id.py]: typo!

@saurabh6790 you need to test this patch very well

File: erpnext_shopify/patches/V1_0/set_variant_id.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+				frappe.db.sql(""" update tabItem set variant_id=shopify_id 
+					where name = %s """, item.get("name"))
+				
+			elif not item.get("has_variants"):
+				try:
+					product = get_request('/admin/products/{}.json'.format(item.get("shopify_id")))['product']
+			
+					frappe.db.sql(""" update tabItem set variant_id=%s 
+						where name = %s """, (product["variants"][0]["id"], item.get("name")))
+				except:
```
Comment (anandpdoshi) [inline erpnext_shopify/patches/V1_0/set_variant_id.py]: Is this acceptable?

File: erpnext_shopify/patches/V1_0/set_variant_id.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+def execute():
+	shopify_settings = frappe.get_doc("Shopify Settings")
+	shopify_items = get_item_list()
+	
+	if shopify_settings.shopify_url and shopify_items:
+		for item in frappe.db.sql("""select name, item_code, shopify_id, has_variants, variant_of from tabItem 
+			where sync_with_shopify=1 and shopify_id is not null""", as_dict=1):
+			
+			if item.get("variant_of"):
+				frappe.db.sql(""" update tabItem set variant_id=shopify_id
```
Comment (anandpdoshi) [inline erpnext_shopify/patches/V1_0/set_variant_id.py]: This fieldname should be `shopify_variant_id`. Also you need to reload Item doctype before running this patch.

**Pull request merged**